blockprod/src/lib.rs

```diff
@@ -145,9 +149,14 @@ mod tests {
 
         tokio::spawn(async move {
             blockprod
-                .call_mut(move |this: &mut Box<dyn BlockProductionInterface>| {
-                    assert!(this.stop_all().is_ok(), "Failed to stop non-existent jobs");
-                    shutdown.initiate();
+                .call_async_mut(move |this| {
+                    Box::pin(async move {
+                        assert!(
+                            this.stop_all().await == Ok(0),
+                            "Failed to stop non-existent jobs"
+                        );
```

Comment:
Not too keen on side effects inside an assert. I'd prefer to run stop_all in a seperate step and let the assert just deal with the result value. Also `assert_eq!` can be used here.

Reply:
Sounds fair. I've moved this out but forgot to clean up the others. I'll clean these all up soon.